cmd/build/helmify/kustomization.yaml

```diff
@@ -79,6 +79,18 @@ patchesJson6902:
         path: /spec/template/spec/nodeSelector/kubernetes.io~1os
       - op: remove
         path: /spec/template/spec/containers/0/image
+      - op: remove
```

Comment:
I think we can get rid of these JSON patches, since these fields will be removed by the Kustomize patch anyway.

Reply:
Hello @maxsmythe ,
I'm actually not sure, if I do remove these one the resulting manifest is not properly formatted.
I did not find a way to just remove the content of `securityContext` with a JSON patch so I had to enumerate all the keys.
Here is the git diff if I remove these from the gatekeeeper-audit-deployment
```diff
+++ b/cmd/build/helmify/kustomization.yaml
@@ -79,18 +79,6 @@ patchesJson6902:
         path: /spec/template/spec/nodeSelector/kubernetes.io~1os
       - op: remove
         path: /spec/template/spec/containers/0/image
-      - op: remove
-        path: /spec/template/spec/containers/0/securityContext/allowPrivilegeEscalation
-      - op: remove
-        path: /spec/template/spec/containers/0/securityContext/capabilities
-      - op: remove
-        path: /spec/template/spec/containers/0/securityContext/readOnlyRootFilesystem
-      - op: remove
-        path: /spec/template/spec/containers/0/securityContext/runAsGroup
-      - op: remove
-        path: /spec/template/spec/containers/0/securityContext/runAsNonRoot
-      - op: remove
-        path: /spec/template/spec/containers/0/securityContext/runAsUser
   - target:
       kind: Deployment
       name: gatekeeper-controller-manager
diff --git a/manifest_staging/charts/gatekeeper/templates/gatekeeper-audit-deployment.yaml b/manifest_staging/charts/gatekeeper/templates/gatekeeper-audit-deployment.yaml
index 542cf226..fffde30b 100644
--- a/manifest_staging/charts/gatekeeper/templates/gatekeeper-audit-deployment.yaml
+++ b/manifest_staging/charts/gatekeeper/templates/gatekeeper-audit-deployment.yaml
@@ -93,6 +93,14 @@ spec:
           {{- toYaml .Values.audit.resources | nindent 10 }}
         securityContext:
           {{- toYaml .Values.audit.securityContext | nindent 10}}
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - all
+          readOnlyRootFilesystem: true
+          runAsGroup: 999
+          runAsNonRoot: true
+          runAsUser: 1000
```

I could eventually do something like that

```diff
diff --git a/cmd/build/helmify/kustomize-for-helm.yaml b/cmd/build/helmify/kustomize-for-helm.yaml
index 00af22bf..5d6c7a95 100644
--- a/cmd/build/helmify/kustomize-for-helm.yaml
+++ b/cmd/build/helmify/kustomize-for-helm.yaml
@@ -167,8 +167,7 @@ spec:
             httpGet:
               path: /healthz
               port: HELMSUBST_DEPLOYMENT_AUDIT_HEALTH_PORT
-          securityContext:
-            HELMSUBST_DEPLOYMENT_AUDIT_SECURITY_CONTEXT: ""
+          HELMSUBST_DEPLOYMENT_AUDIT_SECURITY_CONTEXT: ""
           resources:
             HELMSUBST_DEPLOYMENT_AUDIT_CONTAINER_RESOURCES: ""
       nodeSelector:
diff --git a/cmd/build/helmify/replacements.go b/cmd/build/helmify/replacements.go
index bb8f3a73..e33dc2af 100644
--- a/cmd/build/helmify/replacements.go
+++ b/cmd/build/helmify/replacements.go
@@ -27,7 +27,7 @@ var replacements = map[string]string{

        `HELMSUBST_DEPLOYMENT_AUDIT_AFFINITY: ""`: `{{- toYaml .Values.audit.affinity | nindent 8 }}`,

-       `HELMSUBST_DEPLOYMENT_AUDIT_SECURITY_CONTEXT: ""`: `{{- toYaml .Values.audit.securityContext | nindent 10}}`,
+       `HELMSUBST_DEPLOYMENT_AUDIT_SECURITY_CONTEXT: ""`: `securityContext: {{- toYaml .Values.audit.securityContext | nindent 10}}`,

        `HELMSUBST_DEPLOYMENT_AUDIT_TOLERATIONS: ""`: `{{- toYaml .Values.audit.tolerations | nindent 8 }}`,
```

But then it's not the same approach used for resource requests/limits.

Let me know if I understood your suggestion correctly and how to implement it best.

Reply:
Ah, right, kustomize just sets a field on securityContext.

May as well follow the pre-existing convention. At some point it might be worth considering how to handle cases where a whole sub-object is removed to be redefined.